src/lib/sticky-header/sticky-header-dir.ts

```diff
@@ -0,0 +1,424 @@
+              @Optional() public parentRegion: CdkStickyRegion) {
+    this.element = element.nativeElement;
+    this.upperScrollableContainer = scrollable.getElementRef().nativeElement;
+    this.setStrategyAccordingToCompatibility();
+  }
+
+  ngAfterViewInit(): void {
+    if (!this.isStickyPositionSupported) {
+      this.stickyParent = this.parentRegion != null ?
+        this.parentRegion._elementRef.nativeElement : this.element.parentElement;
+      this.originalCss = this.generateCssStyle(
+        this.getCssValue(this.element, 'zIndex'),
+        this.getCssValue(this.element, 'position'),
+        this.getCssValue(this.element, 'top'),
+        this.getCssValue(this.element, 'right'),
+        this.getCssValue(this.element, 'left'),
+        this.getCssValue(this.element, 'bottom'),
+        this.getCssValue(this.element, 'width'));
+      this.padding = this.getCssValue(this.element, 'padding');
+      this.stickyRegionHeight = this.getCssNumber(this.stickyParent, 'height');
+      this.attach();
+      this.defineRestrictionsAndStick();
+    }
+  }
+
+  ngOnDestroy(): void {
+    if (this._onScrollSubscription) {
+      this._onScrollSubscription.unsubscribe();
+    }
+
+    if (this._onResizeSubscription) {
+      this._onResizeSubscription.unsubscribe();
+    }
+
+    if (this._onTouchSubscription) {
+      this._onTouchSubscription.unsubscribe();
+    }
+  }
+
+  /**
+   * getSupportList() is used to get a list of string which can be set to
+   * sticky-header's style.position and make Sticky positioning work.
+   * It returns a list of string.
+   *
+   * According to the "Position:sticky Browser compatibility" in
+   * "https://developer.mozilla.org/en-US/docs/Web/CSS/position".
+   *
+   * For Desktop: Sticky positioning works well on Chrome, Edge, Firefox and Opera. And can
+   * also work well on Safari with a "-webkit-" prefix. It only does not work on IE.
+   *
+   * For Mobile: Sticky positioning works well on Android Webview, Chrome for Android, Edge,
+   * Firefox Mobile, Opera Mobile. And can also work well on Safari Mobile with a "-webkit-" prefix.
+   * It won't always work on IE phone.
+   *
+   * The implementation references the compatibility checking in Modernizer
+   * (https://github.com/Modernizr/Modernizr/blob/master/feature-detects/css/positionsticky.js).
+   */
+  getSupportList(): Array<string> {
+    let prefixTestList = ['', '-webkit-', '-ms-', '-moz-', '-o-'];
+    let supportList: Array<string> = new Array<string>();
+    let stickyText = '';
+    for (let i = 0; i < prefixTestList.length; i++ ) {
+      stickyText = 'position:' + prefixTestList[i] + 'sticky;';
+      // Create a DOM to check if the browser support current prefix for sticky-position.
+      let div = document.createElement('div');
+      let body = document.body;
+      div.style.cssText = 'display:none;' + stickyText;
+      body.appendChild(div);
+      let isSupport = /sticky/i.test(window.getComputedStyle(div).position);
```

Comment:
We should ensure that this call to `getComputedStyle` happens at the same time as other `getComputedStyle` calls